src/Entity/Reporting/TeamMemberHistory.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace App\Entity\Reporting;
+
+use App\Entity\Adherent;
+use App\Entity\Administrator;
+use App\Entity\Team\Team;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(
+ *     name="team_member_histories",
+ *     indexes={
+ *         @ORM\Index(name="team_member_histories_adherent_id_idx", columns="adherent_id"),
+ *         @ORM\Index(name="team_member_histories_administrator_id_idx", columns="administrator_id"),
+ *         @ORM\Index(name="team_member_histories_date_idx", columns="date")
+ *     }
+ * )
+ * @ORM\Entity
+ */
+class TeamMemberHistory
+{
+    public const ACTION_ADD = 'add';
+    public const ACTION_REMOVE = 'remove';
+
+    public const ACTION_CHOICES = [
+        self::ACTION_ADD,
+        self::ACTION_REMOVE,
+    ];
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(type="integer", options={"unsigned": true})
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var Team
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Team\Team")
+     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     */
+    private $team;
+
+    /**
+     * @var Adherent
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Adherent")
+     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     */
+    private $adherent;
+
+    /**
+     * @var Administrator|null
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Administrator")
+     * @ORM\JoinColumn(nullable=true, onDelete="SET NULL")
+     */
+    private $administrator;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(length=20)
+     */
+    private $action;
+
+    /**
+     * @var \DateTimeImmutable
+     *
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $date;
+
+    private function __construct(Team $team, Adherent $adherent, string $action, ?Administrator $administrator = null)
+    {
+        $this->team = $team;
+        $this->adherent = $adherent;
+        $this->administrator = $administrator;
+        $this->action = $action;
+        $this->date = new \DateTimeImmutable();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getTeam(): Team
+    {
+        return $this->team;
+    }
+
+    public function getAdherent(): Adherent
+    {
+        return $this->adherent;
+    }
+
+    public function getAction(): string
+    {
+        return $this->action;
+    }
+
+    public function getAdministrator(): ?Administrator
+    {
+        return $this->administrator;
+    }
+
+    public function getDate(): \DateTimeImmutable
+    {
+        return $this->date;
+    }
+
+    public function isAdd(): bool
```

Comment:
What about ?
```suggestion
    public function isAdded(): bool
```
And the same for `isRemoved`?